File: engine/engine-model/src/main/scala/tools/packet/wedding/WeddingGiftResult.scala
package tools.packet.wedding

import client.inventory.Item
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class WeddingGiftResult(private var _mode: Byte, private var _itemNames: java.util.List[String],
                        private var _items: java.util.List[Item]) extends PacketInput {
  def mode: Byte = _mode

  def itemNames: java.util.List[String] = _itemNames

  def items: java.util.List[Item] = _items

  override def opcode(): SendOpcode = SendOpcode.WEDDING_GIFT_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/wedding/WeddingGiftResult.java
package tools.packet.wedding;

import java.util.List;

import client.inventory.Item;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record WeddingGiftResult(Byte mode, List<String> itemNames, List<Item> items) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.WEDDING_GIFT_RESULT;
   }
}